File: backend/GraficaModerna.API/Controllers/ProductsController.cs
using GraficaModerna.Application.DTOs;
using GraficaModerna.Application.Interfaces;
using GraficaModerna.Domain.Models;
using Microsoft.AspNetCore.Mvc;

namespace GraficaModerna.API.Controllers;

[Route("api/[controller]")]
[ApiController]
public class ProductsController(IProductService service) : ControllerBase
{
    private readonly IProductService _service = service;

    [HttpGet]
    public async Task<ActionResult<PagedResultDto<ProductResponseDto>>> GetAll(
        [FromQuery] string? search,
        [FromQuery] string? sort,
        [FromQuery] string? order,
        [FromQuery] int page = 1,
        [FromQuery] int pageSize = 8)
    {
        if (page < 1) page = 1;
        if (pageSize > 50) pageSize = 50;

        var result = await _service.GetCatalogAsync(search, sort, order, page, pageSize);
        return Ok(result);
    }

    [HttpGet("{id}")]
    public async Task<ActionResult<ProductResponseDto>> GetById(Guid id)
    {
        var product = await _service.GetByIdAsync(id);
        if (product == null) return NotFound();
        return Ok(product);
    }
}

File: grafica-moderna/app/Http/Controllers/ProductsController.php
<?php

namespace App\Http\Controllers;

use App\Services\ProductService;
use Illuminate\Http\Request;

class ProductsController extends Controller
{
    protected ProductService $service;

    public function __construct(ProductService $service)
    {
        $this->service = $service;
    }

    /**
     * GET /api/products
     */
    public function index(Request $request)
    {
        $search = $request->query('search');
        $sort = $request->query('sort');
        $order = $request->query('order');
        $page = max((int)$request->query('page', 1), 1);
        $pageSize = min((int)$request->query('pageSize', 8), 50);

        $result = $this->service->getCatalog(
            $search,
            $sort,
            $order,
            $page,
            $pageSize
        );

        return response()->json($result);
    }

    /**
     * GET /api/products/{id}
     */
    public function show(string $id)
    {
        $product = $this->service->getById($id);

        if (!$product) {
            return response()->json(['message' => 'Produto não encontrado.'], 404);
        }

        return response()->json($product);
    }
}